### DISCUSSION TITLE
After upgrade on 3.7.1 the proxy over nginx has stopped working

### DISCUSSION THREAD

**Discussion by netskycz** [coder/code-server#2324] [Q&A]
Hello,

after update on 3.7.1 (.rpm package) i got this error:
<img width="1680" alt="Snímek obrazovky 2020-11-17 v 16 05 37" src="https://user-images.githubusercontent.com/27358718/99407564-8777c400-28ef-11eb-8aa6-5b2b8143472a.png">

when i downgrade on 3.6.2 everything is fine.

My nginx:

> location / {
> 	proxy_pass http://127.0.0.1:32765/;
> 	proxy_set_header Host $host;
> 	proxy_set_header Upgrade $http_upgrade;
> 	proxy_set_header Connection upgrade;
> 	proxy_set_header Accept-Encoding gzip;
> }

My ~/.config/code-server/config.yaml

> bind-addr: 127.0.0.1:32765
> auth: password
> password: SOMEPASS
> cert: false
> user-data-dir: /home/code-server/project

I try reconfigure nginx and also config.yaml without success.
Some ideas? Thanks for help.

**Comment by code-asher** [comment]
I gave this a shot but I wasn't able to replicate. Here's what I did:

```
$ docker run -p 8080:80 --name code-server codercom/code-server:3.7.1
$ docker exec code-server sudo apt update
$ docker exec code-server sudo apt install nginx
$ docker cp nginx.conf code-server:/etc/nginx/sites-available/default
$ docker exec code-server sudo nginx
```

Where nginx.conf is:

```
server {
  listen 80 default_server;
  listen [::]:80 default_server;

  location / {
    proxy_pass http://127.0.0.1:8080/;
    proxy_set_header Host $host;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection upgrade;
    proxy_set_header Accept-Encoding gzip;
  }
}
```

Then I went to localhost:8080 which hits the Docker container and NGINX
but everything loaded up for me.

The Docker container uses the .deb though so maybe it could be something
with the .rpm.

Could you post the logs from code-server? It might show us something.

**Reply by code-asher** [reply]
Oh and running code-server with `--log debug` could help.

**Reply by netskycz** [reply]
Today I updated (from 3.6.X) Coder in another environment on Debian (.deb) and the same problem. There is nothing in the debug log (in the case of a proxy). When I access the Coder directly without the Nginx proxy, it works fine (v3.7.1).

Apparently I'm doing something wrong, I'll run a few tests and try to find out where the problem is. I just wanted to ask.

Proxy:
--log debug - empty, only coder standard starting... browser (404 error, go home screen) noting...

Direct:
--log debug - 

```
[2020-11-19T19:07:22.338Z] debug spawned inner process 29920
[2020-11-19T19:07:22.650Z] debug parsed command line {"args":{"_":[],"log":"debug"}}
[2020-11-19T19:07:22.659Z] debug parsed command line {"args":{"_":[],"bind-addr":"IP:PORT","auth":"password","cert":{},"user-data-dir":"/home/code-server/project"}}
[2020-11-19T19:07:22.662Z] debug wrapper 29897 received message from 29920 {"message":{"type":"handshake"}}
[2020-11-19T19:07:22.663Z] debug inner process 29920 received message from 29897 {"message":{"type":"handshake"}}
```
....info ip, config...
```
debug vscode got message from code-server {"type":"init"}
debug vscode got message from code-server {"type":"socket"}
debug vscode got message from code-server {"type":"socket"}
debug vscode New connection {"token":"token"}
```
....everthing ok

I'll check more in the coming days...
